api/src/use_cases/get_config/use_case.rs

```diff
@@ -26,27 +21,34 @@ struct UseCaseImpl {}
 
 #[async_trait]
 impl UseCase for UseCaseImpl {
-    async fn execute(
-        &self,
-        clock: &(dyn Clock + Send + Sync),
-        business_week: BusinessWeek,
-    ) -> Result<Config, Error> {
-        let openapi_business_week: models::BusinessWeek = business_week.into();
-        Ok(Config::new(clock.now().to_string(), openapi_business_week))
+    async fn execute(&self, storage: &(dyn Storage + Send + Sync)) -> Result<Config, Error> {
+        let config = storage
+            .get_config(CONFIG_ID)
+            .await
+            .unwrap()
```

Comment:
the line after it there is unwrap or else 

            .unwrap_or_else(|| {
                // Handle the None case here and return a default value or handle the error gracefully
                println!("Error: Config not found. Returning default config.");
                types::Config::default() // Return default config
            });

Reply:
If my understanding is correct the first `unwrap()` is related to the `Result` which encapsulates the DynamoDB request while the `unwrap_or_else()` is related to the `Option` in the `Result`. In case the DynamoDB request fails, `unwrap_or_else()` won't be reached because `unwrap()` already panics.

Imho we could use `await?` to propagate the error in case of failures instead of `await.unwrap()`.

Reply:
yea but now we need to create a From trait to the error.